### PR TITLE
use fancy assignment operators

### PR DESCRIPTION
**Pull request by nev-r**
htdky5jstrhaerg

with these changes, looks like DIM survives eslint `operator-assignment:always`  unscathed. opinions on enabling it?

### CODE REVIEW

File: src/app/inventory/store/item-index.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ export function createItemIndex(item: DimItem): string {
   // Try to make a unique, but stable ID. This isn't always possible, such as in the case of consumables.
   let index = item.id;
   if (item.id === '0') {
-    _idTracker[index] = (_idTracker[index] || 0) + 1;
+    _idTracker[index] ||= 0;
+    _idTracker[index]++;
```
Comment (bhollis) [inline src/app/inventory/store/item-index.ts:15]: Could also be `(_idTracker[index] ||= 0)++` if you wanted to be evil

File: src/app/item-triage/ItemTriage.tsx [hunk 2]
```diff
@@ -270,7 +270,8 @@ function collectRelevantItemFactors(exampleItem: DimItem, allItems: DimItem[]) {
     .forEach((item: DimItem) => {
       factorCombos[exampleItem.bucket.sort as factorComboCategory].forEach((factorCombo) => {
         const combination = applyFactorCombo(item, factorCombo);
-        combinationCounts[combination] = (combinationCounts[combination] ?? 0) + 1;
+        combinationCounts[combination] ??= 0 + 1;
```
Comment (bhollis) [inline src/app/item-triage/ItemTriage.tsx:273]: This feels like it shouldn't have the +1

**Review by bhollis — COMMENTED** [review]
Seems neat

File: src/app/inventory/store/item-index.ts (hunk 1 above)
Comment (nev-r) [reply src/app/inventory/store/item-index.ts:15]: unfortunately no, you can access a self-editing method of something, like `(thing??=[]).push(`
but increment and assignment operators want something more valid on the left

File: src/app/item-triage/ItemTriage.tsx (hunk 2 above)
Comment (nev-r) [reply src/app/item-triage/ItemTriage.tsx:273]: miscommit, thank you!

**Pull request merged**